file/random_access_file_reader.cc

```diff
@@ -22,6 +22,50 @@
 #include "util/rate_limiter.h"
 
 namespace ROCKSDB_NAMESPACE {
+inline void IOStatsAddBytesIfPositiveByTemperature(Temperature file_temperature,
```

Comment:
I think we can remove "if positive" to make the function name shorter.

Reply:
I'm a little bit worried about performance in this function. Although most of the time this path might involve I/O so performance might not matter, but some users might use something like mmap and I'm worried that we see a performance regression here.

I think to be safe, we might do an if for Temperature::kUnknown and skip everything. We might not need file_io_stats_by_temperature.unknown_file_bytes_read as we always know the total bytes and we can take difference.

By the way, it is not related to PR but IOSTATS_ADD_IF_POSITIVE() is a function I don't understand. Adding a thread local variable by 0 always feels cheaper than doing an if. Also value is almost never 0 so it's a waste. I suggest we just call IOSTATS_ADD() here.

Reply:
Let me create a PR to remove IOSTATS_ADD_IF_POSITIVE(). It doesn't make sense to me at all.

Reply:
Thanks for the review. I will change to IOSTATS_ADD() and remove the `unknown` related counter and skip it at the beginning